### PR TITLE
Feat/2113

### PR DESCRIPTION
**Pull request by Merkur39**
Allow clicking/touching an aggregation line to see the list of corresponding trips.

### CODE REVIEW

File: src/components/Views/Trips.jsx [hunk 1]
```diff
@@ -1,39 +1,78 @@
...
+
+  const trips = useMemo(() => {
+    if (Array.isArray(timeseriesQueryResult)) {
+      return transformTimeseriesToTrips(timeseriesQueryResult)
+    }
+    return []
+  }, [timeseriesQueryResult])
+
+  if (!isLoadingaccountQuery && accounts.length === 0) {
+    return <p>{t('account.notFound')}</p>
```
Comment (trollepierre) [inline src/components/Views/Trips.jsx:59]: nit: at least,  we can test if  there is no data,  we get this account not found translation

File: src/components/Views/ModeAnalysis.jsx [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
 
 const ModeAnalysis = () => {
+  const { t } = useI18n()
+  const history = useHistory()
+  const { mode } = useParams()
   const { isMobile } = useBreakpoints()
 
+  const modeTitle = mode ? t(`trips.modes.${mode}`) : ''
+
+  const onBack = mode ? history.goBack : undefined
```
Comment (trollepierre) [inline src/components/Views/ModeAnalysis.jsx:19]: idem, we could test goBack here

File: src/components/TripsList.jsx [hunk 3]
```diff
@@ -2,79 +2,40 @@ import React, { useMemo } from 'react'
...
       i === 0 || !isSameDay(getStartDate(trip), getStartDate(trips[i - 1])),
     [trips]
   )
 
-  if (isLoading) {
-    return (
-      <Spinner size="xxlarge" className="u-flex u-flex-justify-center u-mt-1" />
-    )
-  }
+  const makeGeojson = useMemo(
```
[the remark this answers is not in the document]
Comment (trollepierre) [reply src/components/TripsList.jsx:17]: in my opinion, we don't need the memoized for a find function

File: src/components/TripsList.jsx [hunk 4]
```diff
@@ -2,79 +2,40 @@ import React, { useMemo } from 'react'
...
-            label={t('loadMore')}
-            fetchMore={tripsQueryResult.fetchMore}
-          />
-        )}
-      </List>
-    </>
+    <List>
+      {trips.map((trip, i) => (
+        <TripItem
+          key={i}
```
Comment (trollepierre) [inline src/components/TripsList.jsx:26]:
```suggestion
          key={trip.id}
```

File: src/components/TripsList.jsx [hunk 5]
```diff
@@ -2,79 +2,40 @@ import React, { useMemo } from 'react'
...
-        {tripsQueryResult.hasMore && (
-          <LoadMore
-            label={t('loadMore')}
-            fetchMore={tripsQueryResult.fetchMore}
-          />
-        )}
-      </List>
-    </>
+    <List>
+      {trips.map((trip, i) => (
```
Comment (trollepierre) [inline src/components/TripsList.jsx:24]:
```suggestion
      {trips.map((trip, index) => (
```

**Review by trollepierre — COMMENTED** [review]
Big progress on testing 👏 congrats.

File: src/components/Views/Trips.jsx [hunk 6]
```diff
@@ -1,39 +1,78 @@
...
+    <>
+      <Titlebar label={t('trips.from') + ' ' + accounts?.[0]?.label} />
+      <TripsList trips={trips} timeseries={timeseriesQueryResult} />
+      {timeseriesQueryLeft.hasMore && (
+        <LoadMore
+          label={t('loadMore')}
+          fetchMore={timeseriesQueryLeft.fetchMore}
+        />
+      )}
+    </>
```
[the remark this answers is not in the document]
Comment (Merkur39) [reply src/components/Views/Trips.jsx:74]: Oui en effet, il y a déjà un travail en cours sur un autre sujet en amont (https://github.com/cozy/coachCO2/issues/61).
Ensuite seulement nous pourrons refactor de manière plus pérenne.

**Review by JF-Cozy — APPROVED** [review]

**Comment by JF-Cozy** [comment]
🎉 👏

**Review by trollepierre — APPROVED** [review]

File: src/components/Analysis/Modes/LoadedModesList.spec.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    }
+  }
+
+  beforeEach(() => {
+    useI18n.mockReturnValue({ t })
+    computeAggregatedTimeseries.mockReturnValue(aggregatedTimeseries)
+    sortTimeseriesByCO2GroupedByMode.mockReturnValue(timeseriesSortedByMode)
+    computeCO2Timeseries.mockReturnValue(48)
+    makeChartProps.mockReturnValue({ data: 'data', options: 'options' })
+    jest.clearAllMocks()
```
[the remark this answers is not in the document]
Comment (Merkur39) [reply src/components/Analysis/Modes/LoadedModesList.spec.js:73]: En effet, il est ben inutile 👍

**Pull request merged**